kimchi/src/circuits/lookup/index.rs

```diff
@@ -36,6 +36,125 @@ pub enum LookupError {
     TableIDZeroMustHaveZeroEntry,
 }
 
+/// Lookup selectors
+#[derive(Clone, Serialize, Deserialize, Debug, Default)]
+pub struct LookupSelectors<T> {
+    /// Chacha pattern lookup selector
+    pub chacha: Option<T>,
+    /// ChachaFinal pattern lookup selector
+    pub chacha_final: Option<T>,
```

Comment:
Maybe,  as part of the (future) gate interface, have each gate define a vector of LookupSelectors (Some for each of the gate's CircuiteGates that has a lookup, otherwise None).  Thus, all the ChaCha lookups could be defined under the "gate".